### PR TITLE
Bloom filter

### PR DESCRIPTION
**Pull request by chaithu14**
Implemented Bloom Filter Operator and also couple of demos.

### CODE REVIEW

File: demos/bloomApp/src/main/resources/com/datatorrent/demos/bloomApp/samplefile.txt [hunk 1]
```diff
@@ -0,0 +1,92 @@
+Mitt Romney’s second presidential run has largely focused on a single theme: Barack Obama has bungled the economy, and I’m the guy who can fix it. Romney points to his business and management expertise, honed in the boardrooms of Bain Capital and showcased by his stewardship of the Salt Lake City Olympics, to argue that his experience creating businesses and turning around flagging ones renders him the best positioned candidate to stanch the economic bleeding and put America back to work. But until now, he hasn’t seemed in too much of a hurry. The Romney campaign has been coasting, assiduously courting donors and opinion-makers, never panicking in grim news cycles and often preferring to plead its case in the op-ed pages instead of on the stump. The candidate has favored studiously casual attire, rarely engaged his rivals, and often skated over the fine print detailing how, exactly, he would engineer a turnaround of the U.S. economy.
```
Comment (chaithu14) [inline demos/bloomApp/src/main/resources/com/datatorrent/demos/bloomApp/samplefile.txt]: Moved the file to resources.

File: demos/bloomApp/src/main/resources/file.txt [hunk 2]
```diff
@@ -0,0 +1,99 @@
+Though
```
Comment (chaithu14) [inline demos/bloomApp/src/main/resources/file.txt]: Moved the file to resources.

File: demos/bloomApp/src/main/java/com/datatorrent/demos/bloomApp/InputOperator.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+ * <p>InputOperator class.</p>
+ *
+ * @since 0.3.2
+ */
+public class InputOperator extends SimpleSinglePortInputOperator<String> implements Runnable {
+
+    private static final Logger logger = LoggerFactory.getLogger(InputOperator.class);
+    protected long averageSleep = 300;
+    protected long sleepPlusMinus = 100;
+    protected String fileName = "com/datatorrent/demos/bloomApp/samplefile.txt";
```
Comment (chaithu14) [inline demos/bloomApp/src/main/java/com/datatorrent/demos/bloomApp/InputOperator.java]: Removed the hard-coded and added NotNull property.

**Comment by andyperlitch** [comment]
Something odd is happening with this merge. The build has gone on since January 17th! Here is the last part of the build log:

```bash
2015-01-17 03:12:16,384 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[98, 57, 55, 98, 49, 52, 101, 100, 54, 55, 102, 50, 52, 49, 48, 97, 97, 48, 99, 52, 51, 54, 97, 98, 52, 51, 54, 49, 55, 48, 49, 48]}, time = 1387123200000
2015-01-17 03:12:16,384 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[102, 99, 97, 51, 48, 100, 101, 52, 99, 52, 52, 99, 52, 57, 98, 55, 57, 56, 56, 101, 97, 100, 98, 101, 52, 101, 98, 52, 52, 54, 98, 101]}, time = 1387123200000
2015-01-17 03:12:16,384 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[102, 49, 54, 51, 48, 98, 101, 99, 57, 102, 52, 98, 52, 98, 101, 49, 98, 101, 101, 97, 55, 49, 56, 99, 102, 56, 56, 100, 57, 52, 57, 50]}, time = 1387123200000
2015-01-17 03:12:16,384 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[53, 55, 101, 102, 52, 100, 55, 48, 48, 102, 99, 54, 52, 98, 57, 101, 57, 54, 54, 55, 53, 55, 53, 57, 57, 57, 48, 55, 53, 57, 50, 54]}, time = 1387123200000
2015-01-17 03:12:16,384 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[56, 101, 55, 97, 56, 52, 99, 49, 51, 55, 97, 50, 52, 55, 57, 48, 98, 97, 99, 99, 97, 50, 97, 102, 50, 51, 99, 99, 49, 50, 52, 100]}, time = 1387123200000
2015-01-17 03:12:16,385 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[57, 97, 101, 102, 56, 98, 51, 53, 100, 54, 56, 57, 52, 98, 55, 54, 56, 50, 50, 99, 51, 97, 54, 100, 97, 49, 53, 98, 55, 53, 56, 98]}, time = 1387123200000
2015-01-17 03:12:16,385 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[55, 101, 57, 57, 50, 52, 52, 54, 55, 99, 97, 51, 52, 99, 54, 97, 57, 52, 56, 53, 48, 100, 97, 53, 101, 99, 56, 100, 98, 102, 101, 50]}, time = 1387123200000
2015-01-17 03:12:16,385 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[57, 101, 101, 99, 57, 54, 101, 98, 48, 52, 56, 101, 52, 56, 98, 100, 57, 102, 53, 99, 54, 100, 55, 48, 97, 97, 56, 52, 101, 48, 100, 53]}, time = 1387123200000
2015-01-17 03:12:16,385 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[98, 100, 102, 48, 97, 102, 51, 101, 52, 102, 102, 97, 52, 55, 101, 55, 98, 51, 102, 55, 100, 57, 51, 52, 53, 97, 102, 55, 52, 98, 100, 48]}, time = 1387123200000
2015-01-17 03:12:16,385 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[49, 98, 98, 50, 99, 98, 56, 48, 52, 49, 51, 53, 52, 97, 50, 97, 98, 57, 48, 51, 98, 99, 98, 98, 56, 51, 50, 55, 56, 55, 100, 52]}, time = 1387123200000
2015-01-17 03:12:16,386 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[102, 99, 48, 57, 54, 51, 48, 98, 51, 48, 57, 57, 52, 49, 100, 51, 57, 56, 51, 98, 49, 101, 102, 49, 98, 53, 102, 56, 49, 97, 52, 100]}, time = 1387123200000
2015-01-17 03:12:16,386 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[98, 53, 97, 55, 57, 50, 57, 100, 98, 98, 100, 100, 52, 98, 99, 49, 97, 102, 49, 50, 52, 49, 57, 55, 50, 50, 101, 51, 101, 51, 48, 50]}, time = 1387123200000
2015-01-17 03:12:16,386 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[57, 50, 50, 50, 101, 102, 55, 54, 55, 97, 99, 99, 52, 48, 54, 48, 97, 102, 97, 53, 52, 48, 98, 100, 48, 51, 50, 99, 102, 100, 49, 99]}, time = 1387123200000
2015-01-17 03:12:16,386 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[48, 97, 49, 99, 101, 49, 97, 49, 49, 55, 56, 97, 52, 51, 57, 50, 97, 55, 48, 100, 57, 51, 49, 102, 55, 56, 52, 102, 97, 101, 101, 55]}, time = 1387123200000
2015-01-17 03:12:16,386 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[54, 50, 49, 99, 57, 97, 49, 51, 100, 97, 100, 54, 52, 97, 53, 49, 57, 57, 54, 100, 50, 50, 99, 57, 53, 100, 99, 102, 102, 56, 54, 54]}, time = 1387123200000
2015-01-17 03:12:16,386 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[100, 101, 99, 53, 54, 51, 48, 52, 97, 101, 50, 98, 52, 99, 57, 49, 97, 102, 97, 50, 52, 54, 50, 98, 50, 100, 52, 50, 98, 57, 53, 54]}, time = 1387123200000
2015-01-17 03:12:16,387 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[97, 56, 99, 101, 54, 102, 57, 98, 100, 56, 49, 54, 52, 54, 53, 57, 57, 51, 53, 102, 101, 98, 54, 56, 97, 99, 101, 53, 56, 98, 56, 55]}, time = 1387123200000
2015-01-17 03:12:16,387 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[101, 51, 99, 56, 101, 52, 99, 53, 98, 53, 57, 101, 52, 101, 99, 55, 56, 48, 53, 57, 53, 49, 51, 54, 99, 56, 52, 101, 53, 100, 99, 55]}, time = 1387123200000
2015-01-17 03:12:16,387 [main] DEBUG interceptor.InterceptorTestHelper processFile - guid = Slice{[50, 48, 51, 51, 50, 52, 50, 102, 55, 53, 52, 51, 52, 49, 51, 52, 56, 101, 55, 49, 54, 100, 49, 54, 100, 102, 51, 54, 54, 98, 48, 102]}, time = 1387123200000
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.478 sec - in com.datatorrent.flume.interceptor.ColumnFilteringInterceptorTest
Running com.datatorrent.flume.sink.ServerTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 sec - in com.datatorrent.flume.sink.ServerTest
Running com.datatorrent.flume.sink.DTFlumeSinkTest
2015-01-17 03:12:16,418 [EventLoop-null] DEBUG sink.DTFlumeSinkTest advertise - listening at Server.Service{id=null, host=aquila12.local, port=37039, payload=null}
```

**Comment by andyperlitch** [comment]
image

**Comment by 243826** [comment]
too old.

**Pull request closed without merging**